core/src/main/java/com/linecorp/armeria/common/http/HttpParameters.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.http;
+
+import static io.netty.util.HashingStrategy.JAVA_HASHER;
+
+import io.netty.handler.codec.DefaultHeaders;
+import io.netty.handler.codec.ValueConverter;
+import io.netty.util.HashingStrategy;
+
+/**
+ * HTTP parameters map which uses a {@link HashingStrategy#JAVA_HASHER}
+ * to support case-sensitive parameter names.
+ */
+public class HttpParameters extends DefaultHeaders<String, String, HttpParameters> {
```

Comment:
Could you split this into interface and implementation like we did for HttpHeaders?